src/EditorFeatures/Core/Shared/Utilities/ProgressTrackerAdapter.cs

```diff
@@ -1,27 +1,39 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using System.Threading;
 using Microsoft.CodeAnalysis.Shared.Utilities;
 using Microsoft.VisualStudio.Utilities;
 
 namespace Microsoft.CodeAnalysis.Editor.Shared.Utilities
 {
     /// <summary>
-    /// An adapter between editor's <see cref="IUIThreadOperationScope"/> (which supports reporting progress) 
-    /// and <see cref="IProgressTracker"/>.
+    /// An adapter between editor's <see cref="IUIThreadOperationScope"/> (which supports reporting
+    /// progress) and <see cref="IProgressTracker"/>.
     /// </summary>
     internal class ProgressTrackerAdapter : IProgressTracker
     {
         private readonly IUIThreadOperationScope _uiThreadOperationScope;
         private int _completedItems;
         private int _totalItems;
+        private string _description;
 
         public ProgressTrackerAdapter(IUIThreadOperationScope uiThreadOperationScope)
```

Comment:
if this already exist, only thing we need is AddScope's scope to create this.

Reply:
This type simply bridges between the two worlds.  The primary problem right now is that IUIThreadOperationScope has a broken implementation.  So you cannot actually do things like show progress or update the message in the UI.  

Also, if you actually try to use ```scope.Context.AddScope``` this causes the UI to flash.  So, basically, there is no good way currently to use the new API to get a good experience here.  When those issues are fixed, i think it woudl be fine to move back onto it.